### PR TITLE
WT-12030 Fortify config_compare in test import base and add named checkpoint test

### PR DESCRIPTION
**Pull request by mariammojid**
- Made changes to config_compare
  - replaced regex functionality with list comprehension
  - created strip_subconfigs helper function to remove the fields that we wont be comparing (id, checkpoints)
- Added named checkpoint test.

### CODE REVIEW

File: test/suite/test_import01.py [hunk 1]
```diff
@@ -77,15 +77,36 @@ def check(self, uri, keys, values):
...
+        subconfigs = []
+        curr_subconfig = []
+        depth = 0
+
+        for char in conf:
+            if char == '(':
+                depth += 1
+            elif char == ')':
+                depth -= 1
+           # If end of one subconfig, append it to subconfigs list.
```
Comment (etienneptl) [inline test/suite/test_import01.py:96]:
```suggestion
           # If we have reached the end of the subconfig, append it to the subconfig list.
```

File: test/suite/test_import01.py [hunk 2]
```diff
@@ -77,15 +77,36 @@ def check(self, uri, keys, values):
...
+            if char == ',' and depth == 0:
+                subconfigs.append(''.join(curr_subconfig))
+                curr_subconfig = []
+            else:
+                # Append char to curr subconfig
+                curr_subconfig.append(char)
+        # If curr not empty append to subconfigs.
```
Comment (etienneptl) [inline test/suite/test_import01.py:103]:
```suggestion
        # Append any subconfig left.
```

File: test/suite/test_import01.py [hunk 3]
```diff
@@ -77,15 +77,36 @@ def check(self, uri, keys, values):
...
+        if curr_subconfig:
+            subconfigs.append(''.join(curr_subconfig))
+
+        sliced_subconfigs = [con for con in subconfigs if not con.startswith("id=") and not con.startswith("checkpoint")]
```
Comment (etienneptl) [inline test/suite/test_import01.py:107]:
```suggestion
        # The ID and checkpoint information can be different between configs, remove it.
        sliced_subconfigs = [con for con in subconfigs if not con.startswith("id=") and not con.startswith("checkpoint")]
```

File: test/suite/test_import01.py [hunk 4]
```diff
@@ -77,15 +77,36 @@ def check(self, uri, keys, values):
     # The ID and checkpoint information can be different between configs, remove it. Everything else
```
Comment (etienneptl) [inline test/suite/test_import01.py:77]: Can we update the comment here to:
```
Compare two given configurations, excluding unique data.
```

File: test/suite/test_import01.py [hunk 5]
```diff
@@ -77,15 +77,36 @@ def check(self, uri, keys, values):
...
-        a = re.sub('id=\d+,?', '', aconf)
-        a = (re.sub('\w+=\(.*?\)+,?', '', a).strip(',').split(',') +
-             re.findall('\w+=\(.*?\)+', a))
-        b = re.sub('id=\d+,?', '', bconf)
-        b = (re.sub('\w+=\(.*?\)+,?', '', b).strip(',').split(',') +
-             re.findall('\w+=\(.*?\)+', b))
-        a = [x for x in a if not x.startswith("checkpoint=")]
-        b = [x for x in b if not x.startswith("checkpoint=")]
-        self.assertTrue(sorted(a) == sorted(b))
+        stripped_aconf = self.strip_subconfigs(aconf)
```
Comment (etienneptl) [inline test/suite/test_import01.py:80]:
```suggestion
        # Retrieve the data that can be compared.
        stripped_aconf = self.strip_subconfigs(aconf)
```

File: test/suite/test_import01.py [hunk 6]
```diff
@@ -77,15 +77,36 @@ def check(self, uri, keys, values):
...
+        stripped_bconf = self.strip_subconfigs(bconf)
+
+        self.assertTrue(sorted(stripped_aconf) == sorted(stripped_bconf))
+
+    # Strips the subconfigs that are irrelevant to config_compare (id and checkpoints).
```
Comment (etienneptl) [inline test/suite/test_import01.py:85]:
```suggestion
    # Remove information related to the ID and checkpoints from a config.
```

File: test/suite/test_import01.py [hunk 7]
```diff
@@ -77,15 +77,36 @@ def check(self, uri, keys, values):
...
+    def strip_subconfigs(self, conf):
```
Comment (etienneptl) [inline test/suite/test_import01.py:86]:
```suggestion
    def strip_subconfig(self, conf):
```

Comment (etienneptl) [reply test/suite/test_import01.py:86]: Please update the two calls after this change.

File: test/suite/test_import01.py [hunk 8]
```diff
@@ -169,6 +190,7 @@ def test_file_import(self):
 
         # Import the file.
         self.session.create(self.uri, import_config)
+        self.session.checkpoint("name=abc")
```
Comment (etienneptl) [inline test/suite/test_import01.py:193]: Please add a comment saying why we are testing this:
```
Create a named checkpoint so it appears in the configuration of the imported table. This should not be relevant when comparing the original configuration and the new one.
```

**Review by etienneptl — COMMENTED** [review]
Really good work! I have left a few comments.

File: test/suite/test_import01.py (hunk 1 above)
Comment (mariammojid) [reply test/suite/test_import01.py:96]: ty, fixed!

File: test/suite/test_import01.py (hunk 2 above)
Comment (mariammojid) [reply test/suite/test_import01.py]: ty, fixed!

File: test/suite/test_import01.py (hunk 3 above)
Comment (mariammojid) [reply test/suite/test_import01.py:108]: ty, fixed!

File: test/suite/test_import01.py (hunk 5 above)
Comment (mariammojid) [reply test/suite/test_import01.py]: ty, fixed!

File: test/suite/test_import01.py (hunk 6 above)
Comment (mariammojid) [reply test/suite/test_import01.py]: ty, fixed!

File: test/suite/test_import01.py (hunk 7 above)
Comment (mariammojid) [reply test/suite/test_import01.py]: yep, done. Thanks for the suggestions :)

File: test/suite/test_import01.py [hunk 9]
```diff
@@ -74,18 +74,40 @@ def check(self, uri, keys, values):
...
+    def strip_subconfig(self, conf):
+        subconfigs = []
+        curr_subconfig = ''
+        depth = 0
+
+        for char in conf:
+            if char == '(':
+                depth += 1
+            elif char == ')':
+                depth -= 1
```
Comment (ruby-oujo) [inline test/suite/test_import01.py:95]: Can we put a line break right after this for readability?

File: test/suite/test_import01.py [hunk 10]
```diff
@@ -74,18 +74,40 @@ def check(self, uri, keys, values):
...
+           # If end of one subconfig, append it to subconfigs list.
+            if char == ',' and depth == 0:
+                subconfigs.append(curr_subconfig)
+                curr_subconfig = ''
+            else:
+                # Append char to curr subconfig
+                curr_subconfig += char
+        # Append any subconfig left.
```
Comment (ruby-oujo) [inline test/suite/test_import01.py:103]: Also a line break right before this comment as well.

File: test/suite/test_import01.py (hunk 9 above)
Comment (mariammojid) [reply test/suite/test_import01.py:95]: yep! done :)

File: test/suite/test_import01.py (hunk 10 above)
Comment (mariammojid) [reply test/suite/test_import01.py:105]: done! ty for the suggestion

**Review by ruby-oujo — APPROVED** [review]
LGTM

File: test/suite/test_import01.py [hunk 11]
```diff
@@ -95,19 +95,19 @@ def strip_subconfig(self, conf):
                 depth -= 1
            # If end of one subconfig, append it to subconfigs list.
             if char == ',' and depth == 0:
-                subconfigs.append(''.join(curr_subconfig))
-                curr_subconfig = []
+                subconfigs.append(curr_subconfig)
+                curr_subconfig = ''
             else:
                 # Append char to curr subconfig
```
Comment (etienneptl) [inline test/suite/test_import01.py:102]:
```suggestion
                # Append the character to the current subconfiguration.
```

**Review by etienneptl — COMMENTED** [review]
Thank you for the last changes, suggested one change related to a comment.

File: test/suite/test_import01.py (hunk 11 above)
Comment (mariammojid) [reply test/suite/test_import01.py]: fixed! ty for suggestion

**Review by etienneptl — APPROVED** [review]
LGTM! Nice work.

**Comment by mariammojid** [comment]
evergreen merge

**Pull request merged**